inject-java/src/test/groovy/io/micronaut/annotation/AnnotateTypeArgSpec.groovy

```diff
@@ -0,0 +1,135 @@
+package io.micronaut.annotation
+
+import io.micronaut.annotation.processing.test.AbstractTypeElementSpec
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.inject.ast.ClassElement
+import io.micronaut.inject.visitor.TypeElementVisitor
+import io.micronaut.inject.visitor.VisitorContext
+import spock.lang.PendingFeature
+
+class AnnotateTypeArgSpec extends AbstractTypeElementSpec {
+
+    void 'is always triggered without annotations'() {
+        when:
+            def introspection = buildBeanIntrospection('addann.AnnotateTypeArg0', '''
+package addann;
+
+import java.util.Map;
+
+class AnnotateTypeArg0 {
+
+ private Map<String, String> nameMap;
+
+}
+
+''')
+        then:
+            introspection.hasAnnotation(MyAnnotation)
+    }
+
+    void 'test annotating 1'() {
+        when:
+            def introspection = buildBeanIntrospection('addann.AnnotateTypeArg1', '''
+package addann;
+
+import javax.validation.constraints.NotBlank;
+import java.util.Map;
+
+class AnnotateTypeArg1 {
+
+ private Map<@NotBlank String, String> nameMap;
+
+}
+
+''')
+        then:
+            introspection.hasAnnotation(MyAnnotation)
+    }
+
+    @PendingFeature(reason = "Annotation processor doesn't trigger for inner classes with type argument only annotation")
+    void 'test annotating 2'() {
```

Comment:
It looks like Javac wouldn't trigger the annotation processing when there is only a type annotation in the inner class.